### ISSUE TITLE
`MaxListenersExceededWarning` when making multiple requests via @sanity/client

### ISSUE THREAD

**Issue by matt-koevort** [sanity-io/get-it#488]
**Describe the bug**

We recently updated our version of `@sanity/client` to v6.20.1, and in doing so started noticing that our systems using it were exhibiting heavy memory leaks. Upon further investigation, we observed that after several calls to `sanityClient.fetch` we'd receive a warning the likes of:
```
(node:27340) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 error listeners added to [TLSSocket]. Use emitter.setMaxListeners() to increase limit
    at genericNodeError (node:internal/errors:984:15)
    at wrappedFn (node:internal/errors:538:14)
    at _addListener (node:events:592:17)
    at TLSSocket.addListener (node:events:610:10)
    at Readable.on (node:internal/streams/readable:1126:35)
    at TLSSocket.once (node:events:654:8)
    at ClientRequest.<anonymous> (<redacted>/node_modules/get-it/src/request/node-request.ts:249:12)
    at Object.onceWrapper (node:events:633:26)
    at ClientRequest.emit (node:events:530:35)
    at ClientRequest.emit (node:domain:488:12)
```
In addition to:
```
(node:24118) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 timeout listeners added to [TLSSocket]. MaxListeners is 10. Use emitter.setMaxListeners() to increase limit
    at genericNodeError (node:internal/errors:983:15)
    at wrappedFn (node:internal/errors:537:14)
    at _addListener (node:events:594:17)
    at TLSSocket.addListener (node:events:612:10)
    at Readable.on (node:internal/streams/readable:1126:35)
    at HTTPParser.parserOnIncomingClient [as onIncoming] (node:_http_client:695:10)
    at HTTPParser.parserOnHeadersComplete (node:_http_common:119:17)
    at TLSSocket.socketOnData (node:_http_client:542:22)
    at TLSSocket.emit (node:events:520:28)
    at addChunk (node:internal/streams/readable:559:12)
```

Looking at the lines from the former, we noticed that it had recently been introduced [here](https://github.com/sanity-io/get-it/blame/main/src/request/node-request.ts#L248-L258). If we roll this back to the version prior to when this change was made, we no longer observe any errors (and I suspect would no longer see any memory leaks).

TL;DR, we are seeing an exceeded limit for both `error` and `timeout` listeners when making multiple requests via Sanity client `fetch`.

**To Reproduce**

I have created a basic node project to reproduce the problem: https://github.com/matt-koevort/sanity-client-memory. Instructions for running the script are provided in the README. Essentially just run `SANITY_PROJECT= SANITY_DATASET= SANITY_TOKEN= npm run start` and observe the errors as seen below.

![image](https://github.com/sanity-io/get-it/assets/35453138/86ed1669-4c07-491a-b2ad-8750ae18893b)

The reproduction makes a sequence of very basic GROQ queries to a given dataset, and should reproduce the warnings.

**Expected behavior**

There shouldn't be memory leak related warnings.

**Which versions of Sanity are you using?**

`@sanity/client@6.20.1`
`get-it@8.6.2`

**What operating system are you using?**

macOS Sonama 14.4.1 (although this problem exists in Linux based containers as well)

**Which versions of Node.js / npm are you running?**

10.7.0
v22.2.0
